### DISCUSSION TITLE
"redirect+启用阻止dns嗅探prevent dns spoofing"会启动自己的udp 53的dns服务。和dnsmasq冲突。提供临时解决方式

### DISCUSSION THREAD

**Discussion by xiaozhezi** [v2rayA/v2rayA#852] [Ideas]
### 版本、安装方式、系统

1. 你在使用什么版本的v2rayA
v2raya 2.0.1
2. 你通过什么方式安装v2rayA
opkg
3. 你所使用的操作系统
mips 红米ac2100路由器

### 描述问题

v2rayA，"redirect+启用阻止dns嗅探prevent dns spoofing"会启动自己的udp 53的dns服务。和dnsmasq冲突

### v2rayA/v2ray运行日志：

```shell
root@OpenWrt:~# tail -f /var/log/v2raya/v2raya.log
2023/02/27 08:47:24.191  [W] [processManager.go:75]  only listen at 127.2.0.17: dns port 53 is occupied by dnsmasq(7986)

```

### v2rayA设置以及相关截图
![image](https://user-images.githubusercontent.com/20010048/225832710-a5883d5c-8eb2-4b52-9ef3-7c3e3abacaa6.png)

### 临时解决方式

方式1：关闭dnsmasq。但关闭dnsmasq，会导致无线终端没法获得ip。
方式2：不关闭dnsmasq，但给dnsmasq配置其他非53端口，例如端口0，则是dnsmasq不开启dns服务。

![image](https://user-images.githubusercontent.com/20010048/225832952-e861e05e-783f-45c2-acc7-75ff6da6f7ad.png)

**Comment by mzz2017** [comment]
这个问题不属于 bug，我将该 issue 转至 discussion。

**Reply by xiaozhezi** [reply]
> 这个问题不属于 bug，我将该 issue 转至 discussion。


这个是bug吧。  v2raya以这种方式配置， 因为和dnsmasq冲突，会导致v2raya启动不起来啊。  

建议优化让v2raya自己的dns服务，自动设置为dnsmasq的上游服务器。

再说了，把这个放到discussion，还有人来看吗？

**Reply by mzz2017** [reply]
v2rayA 不是 systemd，不会做这些事情的。linux 下面配置太复杂，很多事情需要用户自己干涉，软件想做全是很难的，也不一定符合用户预期。

**Comment by qimeila** [comment]
我也遇到的这个问题，我是把openwrt的lan和wan自定义dns都设置成127.2.0.17，想问一下这样是不是就可以解决了。当然直接关闭dnsmasq的dns服务也可以
